### ISSUE TITLE
Ansible-lint returns a `jinja[invalid]` rule violation when using role defaults in task vars in combination with `ansible.utils.ipv4`.

### ISSUE THREAD

**Issue by janhorstmann** [ansible/ansible-lint#4468]
##### Summary

Ansible-lint returns a `jinja[invalid]` rule violation when using role defaults in task vars in combination with `ansible.utils.ipv4`.

It seems that the role default is not taken into account in this case and the variable is undefined. I assume that this also applies to other filters, but have not verified that.

Note that using the role default directly in the task works fine (first task in the reproducer below), as well as providing a default in task vars (second task in the reproducer below).

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

```console
ansible-lint --version
ansible-lint 24.12.3.dev9 using ansible-core:2.18.1 ansible-compat:24.10.0 ruamel-yaml:0.18.10 ruamel-yaml-clib:None
```

- ansible installation method: pip from git branch `main`
- ansible-lint installation method: pip from git branch `main`

##### STEPS TO REPRODUCE

Create a role with the following files and run `ansible-lint` on it.

`defaults/main.yml`
```yaml
---
my_role_default_ipv4_cidr: 127.0.0.1/8
```

`tasks/main.yml`
```yaml
---
- name: No jinja rule violation
  ansible.builtin.debug:
    msg: "{{ my_role_default_ipv4_cidr | ansible.utils.ipv4 }}"

- name: No jinja rule violation
  ansible.builtin.debug:
    msg: "{{ my_ipv4_cidr }}"
  vars:
    my_default_ipv4_cidr: "127.0.0.1/8"
    my_ipv4_cidr: "{{ my_default_ipv4_cidr | ansible.utils.ipv4 }}"

- name: Jinja rule violation
  ansible.builtin.debug:
    msg: "{{ my_ipv4_cidr }}"
  vars:
    my_ipv4_cidr: "{{ my_role_default_ipv4_cidr | ansible.utils.ipv4 }}"

```

##### Desired Behavior

Ansible-lint should take the role default into account when it is used in task vars and subsequently no linting errors should occur.

##### Actual Behavior

An unhandled exception occurs, because the ipv4 filter does not handle `ansible.template.native_helpers.AnsibleUndefined` objects.

```console
ansible-lint .
WARNING  Listing 1 violation(s) that are fatal
jinja[invalid]: An unhandled exception occurred while templating '{{ my_role_default_ipv4_cidr | ansible.utils.ipv4 }}'. Error was a <class 'ansible.errors.AnsibleFilterError'>, original message: Unrecognized type <<class 'ansible.template.native_helpers.AnsibleUndefined'>> for ipv4 filter <value>
tasks/main.yml:15 Task/Handler: Jinja rule violation

Read documentation for instructions on how to ignore specific rule violations.

# Rule Violation Summary

  1 jinja profile:basic tags:formatting

Failed: 1 failure(s), 0 warning(s) on 3 files. Last profile that met the validation criteria was 'min'.
```

**Comment by ssbarnea** [comment]
Looks like a legit bug. Can you try to make a PR to address it?

**Comment by janhorstmann** [comment]
> Looks like a legit bug. Can you try to make a PR to address it?

sorry @ssbarnea , but I do not have time to look into this any time soon. I was hoping that this would be an easy fix for someone with knowledge in the ansible-lint code base.

As a workaround one can provide the defaults again using the `derfaults()` filter

**Comment by ssbarnea** [comment]
Closing as I was not able to reproduce using current release, v25.1.2
